diff --git a/upnpsoap.c b/upnpsoap.c
index b22e2a5..fc8bb18 100644
--- a/upnpsoap.c
+++ b/upnpsoap.c
@@ -1161,26 +1161,32 @@ callback(void *args, int argc, char **argv, char **azColName)
 		{
 			/* Video and audio album art is handled differently */
 			if( *mime == 'v' && (passed_args->filter & FILTER_RES) && !(passed_args->flags & FLAG_MS_PFS) ) {
+				char *attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_TN\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 0);
 				append_with_attributes(str, "protocolInfo=\"http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN\"", dlna_buf, "res");
-				append_with_attributes(str, "dlna:profileID=\"JPEG_TN\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attribute, dlna_buf, "upnp:albumArtURI");
 
+				attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_SM\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 1);
 				append_with_attributes(str, "protocolInfo=\"http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_SM\"", dlna_buf, "res");
-				append_with_attributes(str, "dlna:profileID=\"JPEG_SM\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attributes, dlna_buf, "upnp:albumArtURI");
 
+				attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_MED\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 2);
 				append_with_attributes(str, "protocolInfo=\"http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_MED\"", dlna_buf, "res");
-				append_with_attributes(str, "dlna:profileID=\"JPEG_MED\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attribute, dlna_buf, "upnp:albumArtURI");
 			} else if( passed_args->filter & FILTER_UPNP_ALBUMARTURI ) {
+				char *attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_TN\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 0);
-				append_with_attributes(str, "dlna:profileID=\"JPEG_TN\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attribute, dlna_buf, "upnp:albumArtURI");
 
+				attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_SM\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 1);
-				append_with_attributes(str, "dlna:profileID=\"JPEG_SM\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attribute, dlna_buf, "upnp:albumArtURI");
 
+				attribute = (passed_args->filter & FILTER_UPNP_ALBUMARTURI_DLNA_PROFILEID) ? "dlna:profileID=\"JPEG_MED\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"" : "";
 				snprintf(dlna_buf, sizeof(dlna_buf), "http://%s:%d/AlbumArt/%s-%d.jpg", lan_addr[passed_args->iface].str, runtime_vars.port, album_art, 2);
-				append_with_attributes(str, "dlna:profileID=\"JPEG_MED\" xmlns:dlna=\"urn:schemas-dlna-org:metadata-1-0/\"", dlna_buf, "upnp:albumArtURI");
+				append_with_attributes(str, attribute, dlna_buf, "upnp:albumArtURI");
 			}
 		}
 		if( (passed_args->flags & FLAG_MS_PFS) && *mime == 'i' ) {
